Data/DataModel/PuzzleStatePerTeam.cs

```diff
@@ -64,5 +64,53 @@ public bool IsSolved
         /// Notes input by the team for this puzzle
         /// </summary>
         public string Notes { get; set; }
+
```

Comment:
I just want to record what the old algorithm was for this, in case we get performance issues with this implementation (though I am hopeful we won't). It computed the time penalty solely by looking at how many wrong guesses were in the previous N minutes (data that the submissions page already has).

        public static async Task<TimeSpan> GetRemainingTemporaryBanDurationAsync(SqlConnection connection, int teamId, int puzzleId)
        {
            DataSet ds = await GetTeamPuzzleSubmissionsAsync(connection, teamId, puzzleId, true);
            DataTable dtIncorrectSubmissions = ds.Tables[0];
            DataView dvIncorrectSubmissions = new DataView(dtIncorrectSubmissions);
            dvIncorrectSubmissions.RowFilter = "[DateTimeSubmitted] >= #" + DateTime.UtcNow.AddMinutes(-1 * TemporaryBanIncorrectGuessPeriod).ToString() + "#";
            dvIncorrectSubmissions.Sort = "[DateTimeSubmitted] DESC";
            if (dvIncorrectSubmissions.Count > 0)
            {
                int overGuesses = dvIncorrectSubmissions.Count - TemporaryBanIncorrectGuessLimit;
                DateTime lastGuess = (DateTime)dvIncorrectSubmissions[0]["DateTimeSubmitted"];
                var penaltyFree = lastGuess.AddMinutes(overGuesses*1);
                var penalty = penaltyFree - DateTime.UtcNow;

                if (penalty > TimeSpan.Zero)
                {
                    return penalty;
                }
            }
            return TimeSpan.Zero;
        }